src/main/java/com/google/devtools/build/lib/rules/genrule/GenRuleBase.java

```diff
@@ -321,6 +326,31 @@ public String lookupVariable(String variableName) throws ExpansionException {
         return expandSingletonArtifact(filesToBuild, "$@", "output file");
       }
 
+      if (variableName.equals("OUT_ROOT_DIR")) {
```

Comment:
How about calling this `$RULEDIR` to be consistent with `$GENDIR` and `$BINDIR`?

Reply:
done